### PR TITLE
 Improved matrix construction  And bounds

### PR DESCRIPTION
**Pull request by Bod9001**
custom solution for bounds , 
CL: [Fix] fixed matrix collision issues
CL: [Fix] fixed matrix construction ( it was funny in some directions due to annoying offset stuff )

### CODE REVIEW

**Comment by corp-0** [comment]
rejoice, codacy is back

**Comment by Aranclanos** [comment]
gonna test this a bit

**Comment by Aranclanos** [comment]
image
the bounds are now half a tile smaller than the actual size of the matrixes

**Comment by Aranclanos** [comment]
Serverside sprites no longer snap their orientation and client sprites direction snap to the wrong place, although the latter was bug preexistant to this PR

File: UnityProject/Assets/Scripts/Core/Chat/ChatEntry.cs [hunk 1]
```diff
@@ -254,18 +254,15 @@ private IEnumerator AnimateStackObject()
...
+				var lastCharacter = characterInfo[messageText.textInfo.characterCount - 1];
 				var charWorld = messageText.transform.TransformPoint(lastCharacter.bottomRight);
 				var newWorldPos = stackObject.transform.position;
 				newWorldPos.x = charWorld.x + 3;
 				stackObject.transform.position = newWorldPos;
 			}
-			else
-			{
-				Logger.LogError("Caught an IndexOutOfRange in ChatEntry.SetStackPos() ln 273", Category.Chat);
-			}
```
Comment (Aranclanos) [inline UnityProject/Assets/Scripts/Core/Chat/ChatEntry.cs:268]: this has nothing to do with matrixes or bounds

**Comment by Aranclanos** [comment]
Sometimes my character gets pushed a tile when rotating, even ended up in a window tile. Serverside, cant reproduce easily seems to be rare

File: UnityProject/Assets/Scripts/Core/Chat/ChatEntry.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Chat/ChatEntry.cs:268]: I was getting annoyed by its so I fixed it

**Comment by Bod9001** [comment]
tile bounds are now that,  if you are on the line then you're in it, 
I don't get why it would cause those bugs to happen, seem more like existing bugs may be? idk

**Comment by ThatDan123** [comment]
Tests are failing and see the codacy stuff

**Comment by Bod9001** [comment]
ok cool will do

**Review by ThatDan123 — APPROVED** [review]

**Review by corp-0 — APPROVED** [review]

**Pull request merged**